pynumdiff/linear_model/_linear_model.py

```diff
@@ -1,36 +1,22 @@
-"""
-This module implements some common finite difference schemes
-"""
-import copy
-import math
-import logging
-import scipy
+import copy, math, logging, scipy
 import numpy as np
+from warnings import warn
 
 from pynumdiff import smooth_finite_difference
 from pynumdiff.finite_difference import first_order as finite_difference
 from pynumdiff.utils import utility
 
-# try:
-#     import pychebfun
-# except ImportError:
-#     pass
-
 try:
     import cvxpy
 except ImportError:
     pass
 
-__friedrichs_kernel__ = utility.__friedrichs_kernel__
```

Comment:
There are many places you denote 'private' functions with double underscores on each side. Double underscores are typically for builtin "magic" functions, like basic indexing and math operators. Last I know the convention was prepending a single underscore for privates. Sphinx still gives those a miss, only building docs for the ones that don't start with an underscore. I've been changing the standard to reflect what I think the community is more used to as I go.

I'm also just removing places where you alias a function to its own name. It's easy enough just to call it with `utility._whatever` later on and save this line.